README.md

```diff
@@ -4,247 +4,148 @@ This backend is currently in development and is not advised for Production workf
 
-#### `django_mongodb_backend.parse_uri(uri, conn_max_age=0, test=None)`
+- **Management Commands**  
+    
+  - Use commands like `migrate`, `makemigrations`, `flush`, `sqlmigrate`, many more.
 
-`parse_uri()` provides a few options to customize the resulting `DATABASES`
-setting, but for maximum flexibility, construct `DATABASES` manually as
-described above.
 
-- Use `conn_max_age` to configure [persistent database connections](
-  https://docs.djangoproject.com/en/stable/ref/databases/#persistent-database-connections).
-- Use `test` to provide a dictionary of [settings for test databases](
-  https://docs.djangoproject.com/en/stable/ref/settings/#test).
+## Future Commitments of Django Backend for MongoDB
 
-Congratulations, your project is ready to go!
+- **Advanced Indexing**  
+    
+  - Support for advanced index types like geospatial, text, and vector search indexes.  
+      
+- **Improved Data Modeling**  
+    
+  - Support ArrayFields containing Embedded Models  
+  - Support Collections with multiple Django Models  
+  - Possible support for additional Django fields such as ImageField
 
-## Notes on Django QuerySets
 
-* `QuerySet.explain()` supports the [`comment` and `verbosity` options](
-  https://www.mongodb.com/docs/manual/reference/command/explain/#command-fields).
+- **Extended Querying Features**  
+    
+  - Exploring smoother ways to allow users to use full-text search, vector search, or geospatial querying.
 
-   Example: `QuerySet.explain(comment="...", verbosity="...")`
 
-   Valid values for `verbosity` are `"queryPlanner"` (default),
-   `"executionStats"`, and `"allPlansExecution"`.
+- **Enhanced Transactions Support**  
+    
+  - Investigation and support for transactions, allowing features like `ATOMIC_REQUESTS` and `AUTOCOMMIT`.
 
-## Known issues and limitations
+- **Asynchronous Capabilities**  
+    
+  - Evaluation and support for Django’s asynchronous callback functions.
 
-- The following `QuerySet` methods aren't supported:
-  - `bulk_update()`
-  - `dates()`
-  - `datetimes()`
-  - `distinct()`
-  - `extra()`
-  - `prefetch_related()`
 
-- `QuerySet.delete()` and `update()` do not support queries that span multiple
-  collections.
+- **Performance Optimization**  
+    
+  - Focus on performance tuning, especially concerning JOIN operations and ensure competitive performance relative to SQL databases.
 
-- `DateTimeField` doesn't support microsecond precision, and correspondingly,
-  `DurationField` stores milliseconds rather than microseconds.
 
-- The following database functions aren't supported:
-    - `Chr`
-    - `ExtractQuarter`
-    - `MD5`
-    - `Now`
-    - `Ord`
-    - `Pad`
-    - `Repeat`
-    - `Reverse`
-    - `Right`
-    - `SHA1`, `SHA224`, `SHA256`, `SHA384`, `SHA512`
-    - `Sign`
+- **Expanded Third-Party Library Support**  
+    
+  - Vet our backend library works effortlessly with major Django Third-Party solutions
```

Comment:
solutions? I think you mean "packages" or "libraries"

Reply:
"effortlessly" is likely a huge overstatement. I believe it's often going to be a bit awkward to use MongoDB where SQL databases are expected.